tidb-server/main.go

```diff
@@ -662,19 +662,16 @@ func setGlobalVars() {
 	}
 
 	// For CI environment we default enable prepare-plan-cache.
-	plannercore.SetPreparedPlanCache(config.CheckTableBeforeDrop || cfg.PreparedPlanCache.Enabled)
-	if plannercore.PreparedPlanCacheEnabled() {
-		plannercore.PreparedPlanCacheCapacity = cfg.PreparedPlanCache.Capacity
-		plannercore.PreparedPlanCacheMemoryGuardRatio = cfg.PreparedPlanCache.MemoryGuardRatio
-		if plannercore.PreparedPlanCacheMemoryGuardRatio < 0.0 || plannercore.PreparedPlanCacheMemoryGuardRatio > 1.0 {
-			plannercore.PreparedPlanCacheMemoryGuardRatio = 0.1
-		}
-		plannercore.PreparedPlanCacheMaxMemory.Store(cfg.Performance.ServerMemoryQuota)
-		total, err := memory.MemTotal()
-		terror.MustNil(err)
-		if plannercore.PreparedPlanCacheMaxMemory.Load() > total || plannercore.PreparedPlanCacheMaxMemory.Load() <= 0 {
-			plannercore.PreparedPlanCacheMaxMemory.Store(total)
-		}
+	if config.CheckTableBeforeDrop {
+		plannercore.SetPreparedPlanCache(true)
+	}
```

Comment:
This will be reverted as soon as the sysvarcache is regenerated :(

plannercore.SetPreparedPlanCache() calls variable.PreparedPlanCacheSize.Store() without using the SQL interface.

Reply:
It's only for tests, so it seems OK to use `SetPreparedPlanCache` here. I leave a comment there and plan to remove `SetPreparedPlanCache` in the next Sprint.